src/mono/mono/mini/aot-compiler.c

```diff
@@ -11646,11 +11646,15 @@ emit_class_name_table (MonoAotCompile *acfg)
 	name_p = name_buf = (guint8 *)g_malloc0 (name_buf_size);
 #endif
 
+	guint table_len = table->len;
+	if (table_size > 65000 || table->len > 65000) {
```

Comment:
Why 65000? (As opposed to 65534 or something)

Do we know how much over the limit the microsoft graph API is?  I guess metadata tokens are effectively 24 bits right? so there could be as many as 2^24-1 classes in that assembly?

Reply:
65000 is used to avoid having to think about off by one errors. The assembly has about 40k classes, maybe linking is turned off for it.